### PR TITLE
feat(uneditable): Adds linting for editable dashboards

### LINKED ISSUE #151: Check Idea - Editable: false
Hi 👋 , 

As an idea for a check, that editable is false, because if we are creating our dashboards as code surely the code object should be the source of truth and not overridden? 

It can always be disabled in grafana when making purposeful amendments after all

### PR DESCRIPTION
**Pull request by xorima**
# Description

Today this linter tests a number of settings, but it does not validate that the dashboard cannot be edited after it is deployed in the UI. If this linter is to ensure the source of truth for linted dashboards are a json code object, we should ensure they are not editable by default after deployment.

## Issues Resolved

resolves #151 

## Check List

- [x] New functionality includes testing.

### CODE REVIEW

File: lint/rule_uneditable.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package lint
+
+func NewUneditableRule() *DashboardRuleFunc {
+	return &DashboardRuleFunc{
+		name:        "uneditable-dashboard",
+		description: "Checks that the dashboard is not editable.",
+		fn: func(d Dashboard) DashboardRuleResults {
+			r := DashboardRuleResults{}
+			if d.Editable {
```
Comment (rgeyer) [inline lint/rule_uneditable.go:9]: Editable needs to be added as a field to the `Dashboard` struct in `lint.go`

File: lint/rule_uneditable_test.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewUneditableRule(t *testing.T) {
+	linter := NewUneditableRule()
```
Comment (rgeyer) [inline lint/rule_uneditable_test.go:11]: The new rule needs to be added to the default ruleset in `rules.go`

**Review by rgeyer — CHANGES_REQUESTED** [review]
Thank you for the contribution, and for adding an autofix function also.

This needs a couple small tweaks in order to compile, and for the new rule to be part of the default ruleset.

Once those changes are made, I'll be happy to merge this PR.

Thank you again!

**Comment by xorima** [comment]
@rgeyer I've pushed up the fixes as requested. 

Note also golangci is having some warnings so raised #161  for it and may throw in a pr later today to do as it recomments

**Comment by xorima** [comment]
@rgeyer @tomwilkie any updates on this one?

**Review by rgeyer — APPROVED** [review]

**Comment by rgeyer** [comment]
> @rgeyer @tomwilkie any updates on this one?

Sorry for the unacceptably long delay here, getting this merged now.

**Pull request merged**